attacker/metrics.go

```diff
@@ -73,10 +73,10 @@ func newMetrics(m *vegeta.Metrics) *Metrics {
 		Latencies: LatencyMetrics{
 			Total: m.Latencies.Total,
 			Mean:  m.Latencies.Mean,
-			P50:   m.Latencies.P50,
-			P90:   m.Latencies.P90,
-			P95:   m.Latencies.P95,
-			P99:   m.Latencies.P99,
+			P50:   m.Latencies.Quantile(0.50),
+			P90:   m.Latencies.Quantile(0.90),
+			P95:   m.Latencies.Quantile(0.95),
+			P99:   m.Latencies.Quantile(0.99),
```

Comment:
I'm running into an issue about data races when running it at a high rate. There seems to be a data race between calculating the quartiles for the chart and calculating them for metrics reporting.
![screenshot-1](https://user-images.githubusercontent.com/19730728/96355894-653d2b80-1122-11eb-9ac5-db1cb811de30.png)

It's preferable to re-use a single quantile, but if it gets too complicated, you may want to revert back to it.

```suggestion
			P50:   m.Latencies.P50,
			P90:   m.Latencies.P90,
			P95:   m.Latencies.P95,
			P99:   m.Latencies.P99,
```

Reply:
I'm wondering if it's a little too heavy to use the `sync.Mutex` every time, but I haven't been able to confirm it yet. I'll check it in a few hours.

Reply:
Interesting data race. I'll have a look at it.

From what I gathered, commenting out sendMetrics actually fix that data race but that is not good. I think a mutex, in the case for performance will actually be OK, but it'll be all around the code:

- newMetrics
- adding metrics/calculating quantiles
- redrawChart, as it accesses Latencies to write metrics


Also, I just hit another data race, which happens with commented sendMetrics + starting a new attack:

``` go
WARNING: DATA RACE
Read at 0x00c000190718 by goroutine 20:
  github.com/nakabonne/ali/gui.attack()
      /home/brenol/goworkspace/src/github.com/brenol/ali/gui/keybinds.go:50 +0x5b
  github.com/nakabonne/ali/gui.keybinds.func3()
      /home/brenol/goworkspace/src/github.com/brenol/ali/gui/keybinds.go:40 +0x1d7
  github.com/mum4k/termdash.(*termdash).subscribers.func4()
      /home/brenol/goworkspace/pkg/mod/github.com/mum4k/termdash@v0.12.2/termdash.go:236 +0x7b
  github.com/mum4k/termdash/private/event.(*subscriber).callback()
      /home/brenol/goworkspace/pkg/mod/github.com/mum4k/termdash@v0.12.2/private/event/event.go:95 +0x5a
  github.com/mum4k/termdash/private/event.(*subscriber).run()
      /home/brenol/goworkspace/pkg/mod/github.com/mum4k/termdash@v0.12.2/private/event/event.go:110 +0xd2

Previous write at 0x00c000190718 by goroutine 52:
  github.com/nakabonne/ali/gui.(*drawer).redrawChart()
      /home/brenol/goworkspace/src/github.com/brenol/ali/gui/drawer.go:86 +0x13e8
```

Reply:
OK, changed to a Mutex and fixed both data races. (the one you found and the one I found after)

I did not see a performance hit, but I did not use a lot of workers though.